src/main/Manager.java

```diff
@@ -1,10 +1,92 @@
 package main;
 
 import main.Enums.ActionResult;
+import utils.FileManager;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+
+class Manager extends main.User {
+    ActionResult register(main.User user) {
+        //System.out.println(user.lastLoginDate);
+
+        String result_of_search = "not_ok";
+
+        BufferedReader reader;
+        try {
+            reader = new BufferedReader(new FileReader(
+                    "src/resources/Usernames-and-Passwords"));
+            String line = reader.readLine();
+
+
+            while (line != null) {
+                // System.out.println(line);
+                // read next line
+
+                String ins1 = line;
+                String[] ins2 = ins1.split(",");
+                String username2 = ins2[3];
+                //String password2 = ins2[4];
+                if(user.userName.equals(username2)){
+                    result_of_search = "ok";
+                    break;
+
+                }
+
+                line = reader.readLine();
+            }
+
+            reader.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        if(result_of_search == "ok"){
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [ES: Comparison of String objects using == or != (ES_COMPARING_STRINGS_WITH_EQ)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#es-comparison-of-string-objects-using-or-es-comparing-strings-with-eq).
This code compares java.lang.String objects for reference equality using the == or != operators. Unless both strings are either constants in a source file, or have been interned using the String.intern() method, the same string value may be represented by two different String objects. Consider using the equals(Object) method instead.